File: sequencing.pl
#!/usr/bin/perl
use strict;
use warnings;
my $fastafile = $ARGV[0];
my $flagment = $ARGV[1];
my $seqname = "";
my $seqread = "";
my $pos = -1;
my @array = ();
open(IN,$fastafile) or die "Can't open $fastafile: $!\n";
while(<IN>){
	chomp;
	if(/^>/){
		if($seqname){
			$pos = index($seqread,$flagment);
		}
		while($pos >= 0){
			push(@array,$pos + 1);
			$pos = index($seqread,$flagment,$pos + 1);
		}
		if(@array){
			print "$seqname\n";
			foreach my $ele (@array){
				print "$ele\n";
			}
		}
		$seqname = $_;
		$seqread = "";
		@array = ();
	}elsif(/^[A-Z]/){
		$seqread .= $_;
	}
}
$pos = index($seqread,$flagment);
while($pos >= 0){
	push(@array,$pos + 1);
	$pos = index($seqread,$flagment,$pos + 1);
}
if(@array){
	print "$seqname\n";
	foreach my $ele (@array){
		print "$ele\n";
	}
}
close(IN)

File: sequencing.py
#!/usr/bin/python
import sys
def main():
	fastaFile = open(sys.argv[1],'r')
	flagment = sys.argv[2]
	#init
	seqName = ""
	seqRead = ""
	bPos	= -1
	pArray	= []
	for line in fastaFile:
		line = line.strip("\n")
		if len(line) == 0:
			continue
		elif line[0] == ">":
			#process last read
			if len(seqName) != 0:
				bPos = seqRead.find(flagment)
			while bPos >= 0:
				pArray.append(bPos + 1)
				bPos = seqRead.find(flagment, bPos + 1)
			if len(pArray) != 0:
				print seqName
				for ele in pArray:
					print ele
			#init
			bPos	= -1
			seqName = line
			seqRead = ""
			pArray = []
		else:
			seqRead += line
	bPos= seqRead.find(flagment)
	while bPos >= 0:
		pArray.append(bPos + 1)
		bPos = seqRead.find(flagment, bPos + 1)
	if len(pArray) != 0:
		print seqName
		for ele in pArray:
			print ele
	fastaFile.close()

if __name__ == "__main__":
	main()
